src/D2L.Bmx/Aws/AwsClient.cs

```diff
@@ -13,18 +13,26 @@ internal class AwsClient( IAmazonSecurityTokenService stsClient ) : IAwsClient {
 		AwsRole role,
 		int durationInMinutes
 	) {
-		var authResp = await stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest {
-			PrincipalArn = role.PrincipalArn,
-			RoleArn = role.RoleArn,
-			SAMLAssertion = samlAssertion,
-			DurationSeconds = durationInMinutes * 60,
-		} );
 
-		return new AwsCredentials(
-			SessionToken: authResp.Credentials.SessionToken,
-			AccessKeyId: authResp.Credentials.AccessKeyId,
-			SecretAccessKey: authResp.Credentials.SecretAccessKey,
-			Expiration: authResp.Credentials.Expiration.ToUniversalTime()
-		);
+		try {
+			var authResp = await stsClient.AssumeRoleWithSAMLAsync( new AssumeRoleWithSAMLRequest {
+				PrincipalArn = role.PrincipalArn,
+				RoleArn = role.RoleArn,
+				SAMLAssertion = samlAssertion,
+				DurationSeconds = durationInMinutes * 60,
+			} );
+
+			return new AwsCredentials(
+				SessionToken: authResp.Credentials.SessionToken,
+				AccessKeyId: authResp.Credentials.AccessKeyId,
+				SecretAccessKey: authResp.Credentials.SecretAccessKey,
+				Expiration: authResp.Credentials.Expiration.ToUniversalTime()
+			);
+		} catch( Exception ex ) {
+			if( ex.Message == "The requested DurationSeconds exceeds the MaxSessionDuration set for this role." ) {
+				throw new BmxException( "Duration exceeds the MaxSessionDuration for this role. Lower it in config/parameter", ex );
+			}
+			throw new BmxException( "Error AWS failed to grab credentials", ex );
```

Comment:
we can rephrase it better? thinking:

"Failed to get credentials from AWS"
or
"Error getting credentials from AWS"